packages/blockfrost/src/blockfrostWalletProvider.ts

```diff
@@ -306,12 +248,38 @@ export const blockfrostProvider = (options: Options, logger = dummyLogger): Wall
     ];
   };
 
+  const fetchJsonMetadata = async (txHash: Cardano.Hash16): Promise<Cardano.MetadatumMap | null> => {
+    try {
+      const response = await blockfrost.txsMetadata(txHash);
+      return response.reduce((map, metadatum) => {
+        // Not sure if types are correct, missing 'label', but it's present in docs
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const { json_metadata, label } = metadatum as any;
+        if (!json_metadata || !label) return map;
+        // not sure how Blockfrost represents integer metadatum. Might need to convert to bigint here.
```

Comment:
Are we missing a test to cover this? We can dig up an example tx on _testnet_ that includes a large integer to use as a mock.

Reply:
Yep, we are missing a test for this. That would be nice, how does one dig up tx'es?